src/loader/bgw_launcher.c

```diff
@@ -646,7 +676,7 @@ ts_bgw_cluster_launcher_main(PG_FUNCTION_ARGS)
 
 	CHECK_FOR_INTERRUPTS();
 
-	while (true)
+	while (shutdown_signal == NO_SIGNAL)
```

Comment:
Can't we call `CHECK_FOR_INTERRUPTS` the first thing we do in the loop and then you can just replace the two calls (above and at the end of the loop)  with one.